### PR TITLE
Added exceptions for my dev server

### PR DESCRIPTION
**Pull request by IlyaMenkov**

### CODE REVIEW

File: my_dev_server/tests/api/test_users.py [hunk 1]
```diff
@@ -78,9 +119,31 @@ def test_create_delete_ssh(self):
...
+        self.assertEqual(200, get_ssh.status_code)
+        get_ssh = get_ssh.json()
+        self.assertEqual(ssh['id'], get_ssh['id'])
+        self.assertEqual(ssh['alias'], get_ssh['alias'])
+        self.assertEqual(ssh['username'], get_ssh['username'])
+        self.assertEqual(ssh['host'], get_ssh['host'])
+
+        self._delete_ssh(user, ssh)
+        self._delete_user(user)
+
```
Comment (zemuvier) [inline my_dev_server/tests/api/test_users.py]: Too many blank spaces here

File: my_dev_server/server/main.py [hunk 2]
```diff
@@ -120,7 +132,35 @@ def ssh_delete(user_id, ssh_id):
     models.Ssh.query.filter_by(user_id=user_id,
                                id=ssh_id).delete()
     base.session.commit()
-    return '200'
+    return 'Ssh was deleted successfully', 200
```
Comment (IlyaMenkov) [inline my_dev_server/server/main.py]: just `return`

File: my_dev_server/server/main.py [hunk 3]
```diff
@@ -120,7 +132,35 @@ def ssh_delete(user_id, ssh_id):
...
+
+
```
Comment (IlyaMenkov) [inline my_dev_server/server/main.py]: I need to understand how to handle exceptions in exceptions.py

File: my_dev_server/utils.py [hunk 4]
```diff
@@ -16,8 +16,8 @@
 from oslo_utils import uuidutils
 
 
-def rand_name(name=''):
-    rand_data = uuidutils.generate_uuid()[:8]
+def rand_name(name='', length=8):
```
Comment (IlyaMenkov) [inline my_dev_server/utils.py]: why do we need it?
mb for improving tests?
IDK (

File: my_dev_server/server/common.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from my_dev_server.server import exceptions
+
+MINIMAL_PASSWORD_LEN = 5
+MINIMAL_USERNAME_LEN = 2
+MINIMAL_EMAIL_LEN = 4
```
Comment (esikachev) [inline my_dev_server/server/common.py]: yes, I think config is ok

File: my_dev_server/server/common.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def check_user_correct(username, email, password):
```
Comment (esikachev) [inline my_dev_server/server/common.py]: let's make username=None etc, because if you want to check only one parameter you get the error

**Comment by esikachev** [comment]
@IlyaMenkov also, please, add missed exeption after rebase

File: my_dev_server/server/main.py (hunk 2 above)
Comment (IlyaMenkov) [inline my_dev_server/server/main.py]: It doesn't work (returns 500)

File: my_dev_server/tests/api/base.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import requests
+import testtools
+
+from my_dev_server import utils
+
+
+class TestBase(testtools.TestCase):
```
Comment (esikachev) [inline my_dev_server/tests/api/base.py]: let's use Base for class name

File: my_dev_server/tests/api/test_ssh.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
+import requests
+
+from my_dev_server.tests.api.base import TestBase
```
Comment (esikachev) [inline my_dev_server/tests/api/test_ssh.py]: let's import only modules, not classes

File: my_dev_server/server/main.py [hunk 9]
```diff
@@ -40,13 +54,13 @@ def user_create():
...
         LOG.error(error_msg)
-        return error_msg
+        raise exceptions.Duplicate(error_msg)
 
     if email_exist:
         error_msg = "%s with email: %s" % (USER_EXIST_MSG,
                                            request.json["email"])
         LOG.error(error_msg)
-        return error_msg
+        raise exceptions.Duplicate(error_msg)
```
Comment (zemuvier) [inline my_dev_server/server/main.py]: Duplication of code

File: my_dev_server/tests/api/test_users.py [hunk 10]
```diff
@@ -40,47 +25,128 @@ def test_create_delete_user(self):
...
+          - create user with not correct username
+          - check that exit code and message are correct
+          - create user with not correct password
+          - check that exit code and message are correct
+          - create user with not correct email
+          - check that exit code and message are correct
+        """
+        url = self.url + '/users'
+        data = {
+            "username": 'un',
```
Comment (esikachev) [inline my_dev_server/tests/api/test_users.py]: let's use rnad_name with length=2

File: my_dev_server/tests/api/test_users.py [hunk 11]
```diff
@@ -40,47 +25,128 @@ def test_create_delete_user(self):
...
+            "password": 'pass'
+        }
+        message = 'Password len required > 5'
+        user = requests.post(url, json=data).json()
+        self.assertEqual(411, user['status_code'])
+        self.assertEqual(message, user['message'])
+
+        data = {
+            "username": utils.rand_name('username'),
+            "email": 'un',
```
Comment (esikachev) [inline my_dev_server/tests/api/test_users.py]: ditto

File: my_dev_server/tests/api/test_users.py [hunk 12]
```diff
@@ -40,47 +25,128 @@ def test_create_delete_user(self):
...
 
-        url = self.url % ('/users/%s/ssh/%s' % (
-            user['id'],
-            new_ssh['id']))
-        request = requests.delete(url)
-        self.assertEqual(200, request.status_code)
+        data = {
+            "username": utils.rand_name('username'),
+            "email": utils.rand_name('email'),
+            "password": 'pass'
```
Comment (esikachev) [inline my_dev_server/tests/api/test_users.py]: ditto

**Pull request merged**